src/mbgl/map/tile_worker.cpp

```diff
@@ -37,7 +37,7 @@ TileWorker::~TileWorker() {
 }
 
 TileParseResult TileWorker::parseAllLayers(std::vector<std::unique_ptr<StyleLayer>> layers_,
-                                           const GeometryTile& geometryTile,
+                                           std::unique_ptr<const GeometryTile> geometryTile,
```

Comment:
by moving the entire unique_ptr, we ensure that it gets destructed when this function returns.